src/avro/src/schema.rs

```diff
@@ -502,7 +502,17 @@ impl Name {
             .into());
         }
 
-        let namespace = complex.string("namespace");
+        let (namespace, name) = if let Some(index) = name.rfind('.') {
+            let namespace = name[..index].to_owned();
+            let new_name = name[index + 1..].to_owned();
+            warn!(
```

Comment:
I think this should only be a warning if they actually conflict, i.e., if the schema specifies a namespace but it is not the same namespace as the one computed from the name.

Otherwise, it is probably not a mistake